packages/shared/src/lib/core/layer-2/actions/getGasFeeForLayer1ToLayer2Transaction.ts

```diff
@@ -6,31 +6,31 @@ import { AssetType } from '../enums'
 import { TransferredAsset } from '../types'
 import { outputHexBytes } from '@core/wallet/api'
 
-export async function getGasFeeForLayer1ToLayer2Transaction(sendFlowParameters: SendFlowParameters): Promise<number> {
+export async function getGasFeeForLayer1ToLayer2Transaction(sendFlowParameters: SendFlowParameters): Promise<bigint> {
     const { destinationNetworkId } = sendFlowParameters ?? {}
 
     if (!destinationNetworkId || (destinationNetworkId && isStardustNetwork(destinationNetworkId))) {
-        return 0
+        return BigInt(0)
     }
 
     const transferredAsset = getTransferredAsset(sendFlowParameters)
     if (!transferredAsset) {
-        return 0
+        return BigInt(0)
     }
 
     try {
         const gasEstimate = await getGasEstimateForIscpCall(destinationNetworkId, sendFlowParameters)
         return gasEstimate
     } catch (err) {
         console.error(err)
-        return FALLBACK_ESTIMATED_GAS[sendFlowParameters.type]
+        return BigInt(FALLBACK_ESTIMATED_GAS[sendFlowParameters.type])
```

Comment:
estimated gas cost can be stored as bigint

Reply:
It's less duplication if we convert it here. We also need it as a number in `canAccountMakeEvmTransaction`.